guides/source/routing.md

```diff
@@ -1164,15 +1164,29 @@ resources to use `photos_path` and `accounts_path`.
 
 NOTE: The `namespace` scope will automatically add `:as` as well as `:module` and `:path` prefixes.
 
-You can prefix routes with a named parameter also:
+#### Parametric Scopes
+
+You can prefix routes with a named parameter:
 
 ```ruby
-scope ':username' do
+scope ':account_id', as: 'account', constraints: { account_id: /\d+/ } do
   resources :articles
 end
 ```
 
-This will provide you with URLs such as `/bob/articles/1` and will allow you to reference the `username` part of the path as `params[:username]` in controllers, helpers, and views.
+This will provide you with paths such as `/1/articles/9` and will allow you to reference the `account_id` part of the path as `params[:account_id]` in controllers, helpers, and views.
+
+It will also generate path and URL helpers prefixed with `account_`, into which you can pass your objects as expected:
+
+```ruby
+account_article_path(@account, @article) # => /1/article/9
+url_for([@account, @article]) # => /1/article/9
+form_with(model: [@account, @article]) # => <form action="/1/article/9" ...>
+```
+
+We [use a constraint](#segment-constraints) to limit the scope to only match ID-like strings.
+
+NOTE: The `:as` and `:constraints` parameters are optional, however without the use of `:as` such approach will result in an error when combined with `url_for` helper directly or methods like `form_with` that use `url_for` internally. For example, Rails won't be able to build a route from `url_for([@account, @article])` unless you specified `:as`.
```

Comment:
Nitpick: in the guides, there is a tendency to overuse callouts like `NOTE:`.  When possible, I think it is better to integrate the text with the surrounding content, to stave off callout fatigue.

One way to do that here might be:

```suggestion
We are [using a constraint](#segment-constraints) to limit the scope to only match ID-like strings. You can change the constraint to suit your needs, or omit it entirely. The `:as` option is also not strictly required, but without it, Rails will raise an error when evaluating `url_for([@account, @article])` or other helpers that rely on `url_for`, such as `form_with`.
```